### PR TITLE
Fixes #20884: package state options method breaks "Packages" techniques with new reporting

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/20884

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: tree/30_generic_methods/package_state_options.cf [hunk 1]
```diff
@@ -40,23 +40,32 @@
 bundle agent package_state_options(name, version, architecture, provider, state, options)
 {
   vars:
-      "class_prefix"                string => "package_state_options_${name}";
+    any::
+      "class_prefix"          string => "package_state_options_${name}";
+      "args"                  slist  => { "${name}", "${version}", "${architecture}", "${provider}", "${state}", "${options}" };
+      "report_param"          string => join("_", args);
+      "full_class_prefix"     string => canonify("package_state_options_${report_param}");
+      "old_class_prefix"      string => string_head("${full_class_prefix}", "1000");
```
Comment (ncharles) [inline tree/30_generic_methods/package_state_options.cf:48]: this is really confusing to set "old_class_prefix" to the previous class_previous while "class_prefix" is the "old_class_prefix"
Can you use "previous_class_prefix" ?

**Comment by VinceMacBuche** [comment]
Commit modified

File: tree/20_cfe_basics/common.cf [hunk 2]
```diff
@@ -114,6 +114,28 @@ body classes classes_generic_two(x,y)
                         };
 }
 
+body classes cancel_classes(y)
+{
+   cancel_repaired => { "promise_repaired_$(y)", "$(y)_repaired", "$(y)_ok", "$(y)_reached", "$(y)_not_kept"
```
[the remark this answers is not in the document]
Comment (amousset) [reply tree/20_cfe_basics/common.cf:119]: plus we already have a tree/30_generic_methods/_classes_cancel.cf method which should do the job

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by amousset** [comment]
OK, merging this PR

**Comment by VinceMacBuche** [comment]
Merged with ugly list

**Pull request closed without merging**